common/models/user.js

```diff
@@ -301,6 +301,25 @@ module.exports = function(User) {
     return fn.promise;
   };
 
+  User.observe('before delete', function(ctx, next) {
+    var AccessToken = ctx.Model.app.models.AccessToken;
```

Comment:
I believe it's possible to configure a subclass of `User` to use a subclass of `AccessToken` instead, in which case `app.models.AccessToken` may not exist.

A better solution is to use relation method `ctx.Model.accessTokens(true, cb)` instead (the first argument is called `condOrRefresh`, by setting it to `true` we bypass any cached results).

However, there is no need to find all related tokens and then delete them one by one. You should use `ctx.Model.accessTokens.destroyAll(cb)` instead.

```js
User.observe('before delete', function(ctx, next) {
  ctx.Model.accessTokens.destroyAll(next);
});
```